crmsh/bootstrap.py

```diff
@@ -104,6 +105,179 @@ def init_qdevice(self):
                 cmds=self.qdevice_heuristics,
                 mode=self.qdevice_heuristics_mode)
 
+    def init_sbd_manager(self):
+        self.sbd_manager = SBDManager(self.sbd_device, self.diskless_sbd)
+
+
+class SBDManager(object):
+    """
+    Class SBDManager is for sbd configuration and sbd service management
+    """
+    sbd_config_dict = {
+            "SBD_PACEMAKER": "yes",
+            "SBD_STARTMODE": "always",
+            "SBD_DELAY_START": "no"
+            }
+
+    def __init__(self, sbd_device=None, diskless_sbd=False):
+        """
+        Parameter sbd_device is provided by '-s' option on init process
+        Parameter diskless_sbd is provided by '-S' option on init process
+        """
+        self.sbd_device = sbd_device
+        self.diskless_sbd = diskless_sbd
+        self._sbd_service_flag = False
+
+    @property
+    def sbd_service_flag(self):
+        return self._sbd_service_flag
+
+    @sbd_service_flag.setter
+    def sbd_service_flag(self, value):
+        """
+        Set flag to enable sbd service on join process
+        """
+        self._sbd_service_flag = value
+
+    def sbd_init(self):
+        """
+        Function sbd_init includes these steps:
+        1. Check local environment
+        2. Get sbd device from options or interactive mode
+        3. Initialize sbd device
+        4. Write config file /etc/sysconfig/sbd
+        """
+        self._check_environment()
+        self._get_sbd_device()
+        if not self.sbd_device and not self.diskless_sbd:
+            return
+        self._initialize_sbd()
+        self._update_configuration()
+        # If process work through here, consider it's ready for enable service
+        self._sbd_service_flag = True
+
+    def configured_sbd_device(self):
+        """
+        Gets currently configured SBD device, i.e. what's in /etc/sysconfig/sbd
+        """
+        conf = utils.parse_sysconfig(SYSCONFIG_SBD)
+        return conf.get("SBD_DEVICE")
+
+    def manage_sbd_service(self):
+        if self._sbd_service_flag:
+            invoke("systemctl enable sbd.service")
+        else:
+            invoke("systemctl disable sbd.service")
+
+    def configure_sbd_resource(self):
+        if self.sbd_device and self.configured_sbd_device():
+            if not invoke("crm configure primitive stonith-sbd stonith:external/sbd pcmk_delay_max=30s"):
+                error("Can't create stonith-sbd primitive")
+            if not invoke("crm configure property stonith-enabled=true"):
+                error("Can't enable STONITH for SBD")
+        elif self.diskless_sbd:
+            if not invoke("crm configure property stonith-enabled=true stonith-watchdog-timeout=5s"):
+                error("Can't enable STONITH for diskless SBD")
+
+    def _get_sbd_device(self):
+        if self.sbd_device:
+            self.sbd_device = self._parse_sbd_device()
```

Comment:
I don't know if it's a good idea to change the value of `sbd_device`, it might create issues. We should maybe create other parameter to store the values after parsing. What do you think?

Reply:
I'm not very clear here @arbulu89 , could you give me an example? Thanks!

Reply:
Trying to change this piece of code like below:
```
def _get_sbd_device(self):
        """
        Get sbd device from options or interactive mode
        """
        dev_list = []
        if self.sbd_devices:
            dev_list = self._parse_sbd_device()
            self._verify_sbd_device(dev_list)
        elif not self.diskless_sbd:
            dev_list = self._get_sbd_device_interactive()
        self.sbd_devices = dev_list
```

Reply:
What I don't like is that the value of `self.sbd_devices` changes in format during the usage of this instance. You might have `sbd1;sbd2;sbd3` at the beginning and suddenly you have `['sbd', 'sbd2', 'sbd3']`. So it's a bit confusing. Usually doing this changes it's a bit risky, as you use the variable different in each situation.
That's why I suggested to have 2 variables like `self.sbd_devices_input` where you have always what the user pases, and `self._sbd_devices` after the transformation.
What do you think?

Reply:
OK, changed as `self.sbd_devices_input` and `self._sbd_devices`